### ISSUE TITLE
Accessories not displayed in Homekit

### ISSUE THREAD

**Issue by Patric-Jean** [grzegorz914/homebridge-denon-tv#299]
System: 

Running as a child-bridge on Homebridge Synology / Homebridge-UI
[11/13/2024, 9:48:38 AM] [homebridge-denon-tv] -------- DEnon --------
[11/13/2024, 9:48:38 AM] [homebridge-denon-tv] Manufacturer: Denon
[11/13/2024, 9:48:38 AM] [homebridge-denon-tv] Model: AVR-X2600H
[11/13/2024, 9:48:38 AM] [homebridge-denon-tv] Zones: 2
[11/13/2024, 9:48:38 AM] [homebridge-denon-tv] Control: Main Zone
[11/13/2024, 9:48:38 AM] [homebridge-denon-tv] Firmware: 0
[11/13/2024, 9:48:38 AM] [homebridge-denon-tv] Api version: 301
[11/13/2024, 9:48:38 AM] [homebridge-denon-tv] Serialnr: xxxxxxx
[11/13/2024, 9:48:38 AM] [homebridge-denon-tv] ----------------------------------

[11/13/2024, 9:48:38 AM] [homebridge-denon-tv] Device: 192.168.1.xxx DEnon, Published as external accessory.
[11/13/2024, 9:48:38 AM] [homebridge-denon-tv] Device: 192.168.1.xxx DEnon, Impulse generator started.
[11/13/2024, 9:48:38 AM] DEnon B505 is running on port xxxxx.

The accessories are displayed in Humebridge-UI (Accessories) but not in Homekit. Followed all steps of Traoubleshooting already several times.

**Comment by grzegorz914** [comment]
Are you added it to Home app?

**Comment by Patric-Jean** [comment]
> Are you added it to Home app?

Yes. Of course. The hub is visible. But without accessories (see screenshot). ![image](https://github.com/user-attachments/assets/0c725098-c21d-4bc4-b212-777131f86a00)

**Comment by grzegorz914** [comment]
You need to add the accessory itself, not the bridge

**Comment by Patric-Jean** [comment]
I am very sorry and thankful for your kind help! This seems stupid, but I have configured the accessories in the plugin and added the subbridge (Denon-tv). How can I then add the accessories separately? I dont see separate codes to add them... 🧐 see screenshot.
![2024-11-13_15-26-45](https://github.com/user-attachments/assets/d15e95e8-c590-4a7b-9f12-71223e863265)

**Comment by grzegorz914** [comment]
Use the same code, https://github.com/grzegorz914/homebridge-denon-tv/wiki#adding-to-homekit

**Comment by Patric-Jean** [comment]
> Use the same code, https://github.com/grzegorz914/homebridge-denon-tv/wiki#adding-to-homekit

Yes. That's what I did. Adding the Child-Bridge as an accessory in HomeKit. But as a result, I see Denon-TV as a bridge in HomeKit without any accessories like power or volume.

**Comment by grzegorz914** [comment]
Do not use special character in accessory name like `-`

**Comment by grzegorz914** [comment]
You need to a `DEnon` accessory, not the `homebridge-denon-tv`

**Comment by Patric-Jean** [comment]
Ok. Turns out, I was able to add it to HomeKit (and displaying the accessories) with my iPad that runs on iPadOS 17.x. I just tried it again with my iPhone running iOS 18.2 and it did not work.

**Comment by Patric-Jean** [comment]
Really appreciate your kind and prompt help! ❤️

**Comment by Patric-Jean** [comment]
Do you have paypal for a small thank you? Github lets me send via credit card only.

**Comment by grzegorz914** [comment]
No, I don’t have PayPal

**Comment by Patric-Jean** [comment]
Ok. Ill figure it out. Will come.

**Comment by JeffGoldner** [comment]
I'm in the same boat now. Accessory went missing with the last plug-in update and not possible to add it back. Removed the receiver from HomeKit and the Denon bridge - which was showing no accessories; restarted Homebridge. I tried a number of things including not using a child bridge, but each attempt to add it back gives me the homebridge setup code and no way to add the child bridge (if using), or the accessory receiver. I've been using this for a number of years.

**Comment by goshawk2929** [comment]
I too have been having issues.  I think it is TVoS 18 issue.  Ever since new apple TVOS, Receiver is added as a "speaker" and acts/functions like a remote in iPhone/iPad and not work within homekit.  Not sure if this is a desired result or not....but I liked having the tiles and controls within homekit

**Comment by goshawk2929** [comment]
Like OP, lately the device does not show up to add in homekit.    Notice the names.....in homekit a completely different device shows up.  
<img width="869" alt="Screenshot 2024-11-15 at 5 34 47 PM" src="https://github.com/user-attachments/assets/3d83be76-8e2e-41bb-8b94-81b486c39c4c">
<img width="503" alt="Screenshot 2024-11-15 at 5 37 36 PM" src="https://github.com/user-attachments/assets/e434fbcf-2d4e-44e9-a75a-c67362f8fb25">

{
    "devices": [
        {
            "name": "Denon AV Receiver OJ",
            "host": "xx.xx.x.x,
            "port": 8080,
            "generation": 1,
            "zoneControl": 0,
            "getInputsFromDevice": true,
            "getFavoritesFromDevice": false,
            "getQuickSmartSelectFromDevice": false,
            "inputsDisplayOrder": 0,
            "buttons": [
                {
                    "displayType": 0
                }
            ],
            "buttonsZ2": [
                {
                    "displayType": 0
                }
            ],
            "buttonsZ3": [
                {
                    "displayType": 0
                }
            ],
            "sensorPower": false,
            "sensorVolume": false,
            "sensorMute": false,
            "sensorInput": false,
            "sensorSurrounds": [
                {
                    "displayType": 0
                }
            ],
            "infoButtonCommand": "MNMEN ON",
            "masterPower": false,
            "masterMute": false,
            "masterVolume": false,
            "volumeControlNamePrefix": false,
            "volumeControl": 2,
            "volumeMax": 100,
            "refreshInterval": 5,
            "enableDebugMode": false,
            "disableLogInfo": false,
            "disableLogDeviceInfo": false,
            "disableLogConnectError": false,
            "restFul": {
                "enable": false,
                "debug": false
            },
            "mqtt": {
                "enable": false,
                "debug": false,
                "auth": false
            }
        }
    ],
    "_bridge": {
        "username": “xx:xx:xx:xx:xx:xx”,
        "port": xxxxx
    },
    "platform": "DenonTv"
}

**Comment by grzegorz914** [comment]
Do not use special char in the name

**Comment by ayybe** [comment]
Also unable to add in ios18, no special characters. Bridge adds fine but only a speaker accessory shows up which can’t add

**Comment by ayybe** [comment]
No ios 17 devices in the house and macOS home app won’t allow adding accessories!

**Comment by grzegorz914** [comment]
Go to wiki and try troubleshooting
